### ISSUE TITLE
Supress validation for a line/file

### ISSUE THREAD

**Issue by jasperfirecai2** [github/vscode-github-actions#123]
**Is your feature request related to a problem? Please describe.**
On some workflows, the linter will mark things as *potential* issues because of the way it was written. however, as a writer you have additional information and can possible conclude that there is no problem. In that situation, having a linter warning is mildly annoying. 

For example, I have a workflow that dynamically gets the name of a secret and the accesses `secrets[that value]`. Linter tell me that the context access *might* be invalid. this is correct, but I know that the only possible values should be valid or I've done something wrong elsewhere.

**Describe the solution you'd like**
I'd like to have something along the lines of #pylint: disable etc. for this extension, whether it be a comment ( similar to pylint or eslint ), a config file (e.g. for file x ignore y), an extension setting (ignore warnings of type/ ignore files). To suppress warnings that you know are invalid, to help reduce false positives for issues that the extension simply lacks context for, or to purge warnings that come from bugs of third party workflow authors.

**Additional context**
related: #59

**Comment by felipesu19** [comment]
https://github.com/github/vscode-github-actions/issues/196

**Comment by CaBazaga** [comment]
> #196

Not the same, although related to same root problem (Underided problems being reported)
it's a diametrically opposed solution covering a very different scenario.

Extended comment there about the difference.

TLDR:

This is about "Configure your code". (Which has its use cases)
The other (#196) is about "configure the linter". (Which covers a very different use cases)

**Comment by jasperfirecai2** [comment]
While my use case is not the same as that of the other issue, the solution to both can be one and the same. A comment can be file-wide and a config file can be file or project wide. Read the 'describe the solution you'd like'

**Comment by CaBazaga** [comment]
> While my use case is not the same as that of the other issue, the solution to both can be one and the same. A comment can be file-wide and a config file can be file or project wide. **Read the 'describe the solution you'd like'**

You're right, my fault was reading the **title**... 😉

**Comment by cringdahl** [comment]
This would be especially useful for something like [octokit/request-action](https://github.com/marketplace/actions/github-api-request), which takes arbitrary input keys, but kicks out errors for every one of them.

**Comment by austinibele** [comment]
Bump. This would be very helpful.

**Comment by MatthieuBizien** [comment]
I had to disable the extension because of this, otherwise I have false positives due to problems in .venv.

Official VSCode response: "I think the right and efficient approach is to extensions provide options to exclude." https://github.com/microsoft/vscode/issues/22289  from @sandy081 

<img width="897" height="167" alt="Image" src="https://github.com/user-attachments/assets/9d2feb40-4e99-4359-9bb9-7f095ba2e54e" />

**Comment by alexkli** [comment]
@MatthieuBizien Agree here:

1. VS Code should have an option to ignore problems in folders globally. I basically never want to see errors from 3rd party code in `node_modules`, regardless of language/linter/extension. Too bad VS code team doesn't see this the same way :-(
2. Without 1 being available, the github actions extension should offer a way to ignore or just an option to respect `.gitignore`, on by default, so I don't have to configure the same folders many times. 🙏 

A polluted problems view filled with false positives that never go away (!) is just going to be useless since you'll be trained to ignore it.

**Comment by aauren** [comment]
Sad that this has gone on for so long without any update. This is pretty basic for almost any linter / language server.
